### PR TITLE
[Messenger] Use `TransportMessageIdStamp` in `InMemoryTransport` allows retrying

### PR DESCRIPTION
**Pull request by alexndlm**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| License       | MIT

### CODE REVIEW

File: src/Symfony/Component/Messenger/Transport/InMemoryTransport.php [hunk 1]
```diff
@@ -75,7 +89,8 @@ public function reject(Envelope $envelope): void
     public function send(Envelope $envelope): Envelope
     {
         $this->sent[] = $envelope;
-        $id = spl_object_hash($envelope->getMessage());
+        $id = microtime().spl_object_hash($envelope->getMessage());
```
Comment (jderusse) [inline src/Symfony/Component/Messenger/Transport/InMemoryTransport.php]: what's about `uniqid('', true)` instead?

Comment (alexndlm) [reply src/Symfony/Component/Messenger/Transport/InMemoryTransport.php]: Good idea. Done.

**Comment by okwinza** [comment]
This seems to add some additional logic and thus feels more like a feature to me and should target 5.4 branch.
Are you sure it's a bug?

**Comment by alexndlm** [comment]
> This seems to add some additional logic and thus feels more like a feature to me and should target 5.4 branch.
> Are you sure it's a bug?

For now, it is not possible to resend the same message with InMemory transport. I think it is a bug.

File: src/Symfony/Component/Messenger/Transport/InMemoryTransport.php [hunk 2]
```diff
@@ -75,7 +89,8 @@ public function reject(Envelope $envelope): void
     public function send(Envelope $envelope): Envelope
     {
         $this->sent[] = $envelope;
-        $id = spl_object_hash($envelope->getMessage());
+        $id = uniqid('', true);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Messenger/Transport/InMemoryTransport.php]: what about a simple integer stored as a property on the instance?

**Review by nicolas-grekas — APPROVED** [review]
Except from a minor comment, this makes ways more sense than the previous logic!

File: src/Symfony/Component/Messenger/Transport/InMemoryTransport.php (hunk 2 above)
Comment (alexndlm) [reply src/Symfony/Component/Messenger/Transport/InMemoryTransport.php]: I did not understand your suggestion :(

**Review by nicolas-grekas — APPROVED** [review]
I've updated your patch to my suggestion

**Comment by nicolas-grekas** [comment]
Thank you @alexndlm.

**Pull request merged**